plutus-core/plutus-ir/src/PlutusIR/Purity.hs

```diff
@@ -16,7 +16,8 @@ module PlutusIR.Purity
 import PlutusCore.Builtin
 import PlutusIR
 
-import Control.Applicative
+import Control.Applicative ((<|>))
+import Data.Foldable (asum)
```

Comment:
Kinda nasty... Looks like we need to add this to the style guide or something as we'll inevitably break it again in the future.

Reply:
I think if we wait a while the issue will go away because 8.12 won't be involved in CI any more. We could change it back when that happens, but we'll probably have forgotten about it by then.